Library/src/Systems/Users/Authentication.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+
+#include "CSP/Systems/Users/Authentication.h"
+
+
+namespace csp::systems
+{
+
+/// @brief Result structure for a logout state request.
+class CSP_API LogoutResult : public NullResult
```

Comment:
These types now all inherit from ResultBase-derived types. The intent is to allow them to do internal processing on the response while not providing their own result values, instead making use of the parent type's result values. This simplifies the public API and makes the return type of a callback more apparent by using `StringResultCallback` instead of `AgoraUserTokenResultCallback`.